src/pal/src/loader/module.cpp

```diff
@@ -1620,6 +1620,8 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
         shortAsciiName = FREEBSD_LIBC;
 #elif defined(__NetBSD__)
         shortAsciiName = "libc.so";
+#elif !HAVE_GNU_LIBNAMES_H
+        shortAsciiName = "libc.musl-x86_64.so";
```

Comment:
It does not seem right to check for musl using `#ifndef __GLIBC__`. It will just cause troubles if there are other LIBCs that do not define `__GLIBC__`.

I see that the musl [does not want to add define to identify it](http://wiki.musl-libc.org/wiki/FAQ#Q:_why_is_there_no_MUSL_macro_). Given this, should we figure the libc name in the config script? There is something similar for FreeBSD already. Config scripts is where this ugly stuff stuff belongs...